### PR TITLE
[5.3] Allow passing only two arguments to date based wheres

### PR DESCRIPTION
**Pull request by themsaid**
Currently we can do `->where('name', 'Jon')` but can't do `->whereYear('created_at', 2016)`

This PR introduces the ability to pass only 2 arguments to date-based-where methods:

`whereDate('created_at', '2016-10-10')`
`whereMonth('created_at', '10')`
`whereDay('created_at', '10')`
`whereYear('created_at', '2016')`

### CODE REVIEW

File: src/Illuminate/Queue/Worker.php [hunk 1]
```diff
@@ -148,7 +148,7 @@ protected function daemonShouldRun()
      */
     protected function waitForChildProcess($processId, $timeout)
     {
-        declare(ticks=1) {
+        declare (ticks = 1) {
```
Comment (themsaid) [inline src/Illuminate/Queue/Worker.php]: @GrahamCampbell this change was made when I ran `php-cs-fixer fix` on my machine, but it breaks on StyleCI.

Comment (GrahamCampbell) [inline src/Illuminate/Queue/Worker.php]: please don't send this unrelated change

Comment (GrahamCampbell) [inline src/Illuminate/Queue/Worker.php]: this is not the correct standard anyway

Comment (GrahamCampbell) [inline src/Illuminate/Queue/Worker.php]: please review the phpdoc on strict types for an example delair statement

Comment (themsaid) [inline src/Illuminate/Queue/Worker.php]: Yes I know, I'm just mentioning that there's a difference now between php fixer and style CI.

Comment (themsaid) [inline src/Illuminate/Queue/Worker.php]: I didn't manually make this change, It was done by running `php-cs-fixer fix`. I'm manually un-doing it now, but just mentioning this because in laravel docs it says running `php-cs-fixer fix` should take care of correcting the code style.

Comment (GrahamCampbell) [inline src/Illuminate/Queue/Worker.php]: Please don't run php-cs-fixer.

Comment (GrahamCampbell) [inline src/Illuminate/Queue/Worker.php]: > but just mentioning this because in laravel docs it says running php-cs-fixer fix should take care of correcting the code style.

Where does it say that. I thought I removed that doc because it was wrong?

Comment (themsaid) [inline src/Illuminate/Queue/Worker.php]: Oh my bad, I read it a while back and didn't notice it was removed. 
Alright won't run php-cs-fixer.

File: src/Illuminate/Database/Query/Builder.php [hunk 2]
```diff
@@ -1082,41 +1078,47 @@ public function orWhereDate($column, $operator, $value)
      * Add a "where day" statement to the query.
      *
      * @param  string  $column
-     * @param  string   $operator
-     * @param  int   $value
-     * @param  string   $boolean
+     * @param  string  $operator
+     * @param  mixed  $value
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Query/Builder.php]: why is this fixed, is int|null not correct?

Comment (GrahamCampbell) [inline src/Illuminate/Database/Query/Builder.php]: mixed is a last resort

Comment (themsaid) [inline src/Illuminate/Database/Query/Builder.php]: I saw `mixed` used in `where()` so followed the practise.

```
/**
     * Add a basic where clause to the query.
     *
     * @param  string|array|\Closure  $column
     * @param  string  $operator
     * @param  mixed   $value
     * @param  string  $boolean
     * @return $this
     *
     * @throws \InvalidArgumentException
     */
    public function where($column, $operator = null, $value = null, $boolean = 'and')
    {
```

**Comment by srmklive** [comment]
@themsaid, the following doesn't seems to be working on latest 5.3.x commit: 

For `whereDate`, `whereMonth`, `whereDay` & `whereYear`, the following code when using MySQL (version 5.6.31):

    // whereDate
    DB::table('users')->whereDate('created_at', '2016-06-17');
    
    // whereMonth
    DB::table('users')->whereMonth('created_at', '6');
    
    // whereDay
    DB::table('users')->whereDay('created_at', '17');

    // whereYear
    DB::table('users')->whereDay('created_at', '2016');    

gives the following SQL output:

    select * from `users` where date(`created_at`) = ?;
    select * from `users` where month(`created_at`) = ?;
    select * from `users` where day(`created_at`) = ?
    select * from `users` where year(`created_at`) = ?

whereas the SQL output should be:

    select * from `users` where date(`created_at`) = '2016-06-17';
    select * from `users` where month(`created_at`) =  '6';
    select * from `users` where day(`created_at`) = '17';
    select * from `users` where year(`created_at`) = '2016';

**Comment by arcanedev-maroc** [comment]
Is this compatible with SQLite ??
